applications/launchpad/gui-react/src/pages/home/index.tsx

```diff
@@ -1,9 +1,43 @@
+import { useCallback, useRef } from 'react'
+
 import MainLayout from '../../layouts/MainLayout'
+import { MiningNodeType } from '../../types/general'
+import { useAppSelector, useAppDispatch } from '../../store/hooks'
+import { selectSchedules } from '../../store/app/selectors'
+import { actions as miningActions } from '../../store/mining'
+import useMiningScheduling from '../../useMiningScheduling'
 
 /**
  * Home page: '/'
  */
 const HomePage = () => {
+  const dispatch = useAppDispatch()
+  const schedules = useAppSelector(selectSchedules)
+  const startPending = useRef<boolean>(false)
+  const startMining = useCallback(async (node: MiningNodeType) => {
+    if (startPending.current) {
+      return
+    }
+
+    try {
+      startPending.current = true
+      await dispatch(miningActions.startMiningNode({ node })).unwrap()
+      startPending.current = false
+    } finally {
+      startPending.current = false
+    }
+  }, [])
+  const stopMining = useCallback(
+    (node: MiningNodeType) => dispatch(miningActions.stopMiningNode({ node })),
+    [],
+  )
+
+  useMiningScheduling({
+    schedules,
+    startMining,
+    stopMining,
+  })
+
```

Comment:
Can we move all code added here to a new file, ie.`useSchedule.ts`?
I'm wondering whether the `App.tsx` wouldn't be a better place for it... Although both looks good to me

Reply:
you're right, rearranged the files a bit and moved the hook App.tsx - that meant I moved PersistGate to index.tsx, but I think that's even better
see last commit